### PR TITLE
Add support for `3DTILES_content_conditional` in serializer / deserializer

### PR DESCRIPTION
**Pull request by lilleyse**
## Description

Adds support for [`3DTILES_content_conditional`](https://github.com/CesiumGS/3d-tiles/pull/834) in `Cesium3DTiles`, `Cesium3DTilesReader`, and `Cesium3DTilesWriter`. This is used by an internal pipeline generating time-dynamic 3D Tiles.

Note: this is still a draft extension, but should be safe to merge as not too many changes are expected.

## Author checklist

- [x] I have submitted a [Contributor License Agreement](https://github.com/CesiumGS/community/tree/main/CLAs) (only needed once).
- [x] I have done a full self-review of my code.
- [x] I have updated `CHANGES.md` with a short summary of my change (for user-facing changes).
- [x] I have added or updated unit tests to ensure consistent code coverage as necessary.
- [x] I have updated the documentation as necessary.

## Reviewer checklist

Thank you for taking the time to review this PR. By approving a PR you are taking as much responsibility for these changes as the author.

As you review, please go through the checklist below:

- [ ] Review and run all parts of the test plan on this branch and verify it matches expectations.
    - [ ] If the issue is a bug please make sure you can reproduce the bug in the main branch and then checkout this branch to make sure it actually solved the issue.
- [ ] Review the code and make sure you do not have any remaining questions or concerns. You should understand the code change and the chosen approach. If you are not confident or have doubts about the code, please do not hesitate to ask questions.
    - [x] Code should follow the [C++ Style Guide](https://cesium.com/learn/cesium-native/ref-doc/style-guide.html)
- [x] Review the unit tests and make sure there are no missing tests or edge cases.
- [x] Review documentation changes and updates to `CHANGES.md` to make sure they accurately cover the work in this PR.
- [x] Verify that the [Contributor License Agreement](https://github.com/CesiumGS/community/tree/main/CLAs) has been submitted, if needed.

### CODE REVIEW

File: tools/generate-classes/package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
...
...": "node index.js --schemas https://raw.githubusercontent.com/CesiumGS/3d-tiles/cesium-native/specification/schema/tileset.schema.json https://raw.githubusercontent.com/CesiumGS/3d-tiles/cesium-native/specification/schema/common/rootProperty.schema.json https://raw.githubusercontent.com/CesiumGS/3d-tiles/cesium-native/specification/schema/PropertyTable/propertyTable.schema.json https://raw.githubusercontent.com/CesiumGS/3d-tiles/cesium-native/specification/schema/Subtree/subtree.schema.json https://raw.githubusercontent.com/CesiumGS/3d-tiles/cesium-native/specification/schema/Schema/schema.schema.json https://raw.githubusercontent.com/CesiumGS/3d-tiles/cesium-native/specification/schema/Statistics/statistics.schema.json https://raw.githubusercontent.com/CesiumGS/3d-tiles/cesium-native/extensions/3DTILES_content_conditional/schema/conditionalContent.schema.json --output ../../Cesium3DTiles --readerOutput ../../Cesium3DTilesReader --writerOutput ../../Cesium3DTilesWriter --extensions https://raw.githubusercontent.com/CesiumGS/3d-tiles/cesium-native/extensions/ --namespace Cesium3DTiles --readerNamespace Cesium3DTilesReader --writerNamespace Cesium3DTilesWriter --config 3dTiles.json",
```
Comment (j9liu) [inline tools/generate-classes/package.json:8]: Is this needed? Could we merge the `3d-tiles/3DTILES_content_conditional` branch into `3d-tiles/cesium-native` to avoid this extra input?

File: Cesium3DTiles/generated/include/Cesium3DTiles/Extension3dTilesContentConditional.h [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  /**
+   * @brief The original name of this type.
+   */
+  static constexpr const char* TypeName = "Extension3dTilesContentConditional";
+  /** @brief The official name of the extension. This should be the same as its
+   * key in the `extensions` object. */
+  static constexpr const char* ExtensionName = "3DTILES_content_conditional";
+
+  /**
+   * @brief dimensions
```
Comment (j9liu) [inline Cesium3DTiles/generated/include/Cesium3DTiles/Extension3dTilesContentConditional.h]: Missing spec comment?

File: Cesium3DTiles/generated/include/Cesium3DTiles/Extension3dTilesContentConditionalDimensionsValue.h [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+  /**
+   * @brief name
+   */
+  std::string name;
+
+  /**
+   * @brief keySet
+   */
+  std::vector<CesiumUtility::JsonValue> keySet;
```
Comment (j9liu) [inline Cesium3DTiles/generated/include/Cesium3DTiles/Extension3dTilesContentConditionalDimensionsValue.h:33]: Missing spec comments?

File: Cesium3DTilesWriter/include/Cesium3DTilesWriter/ConditionalContentWriter.h [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+// forward declarations
+namespace Cesium3DTiles {
+struct ConditionalContent;
+}
+
+namespace Cesium3DTilesWriter {
+
+/**
+ * @brief The result of writing a ConditionalContent with
+ * {@link ConditionalContentWriter::writeConditionalContent}.
```
Comment (j9liu) [inline Cesium3DTilesWriter/include/Cesium3DTilesWriter/ConditionalContentWriter.h]: Prefer `\ref` over `@link` per style guide:

```suggestion
 * @brief The result of writing a \ref ConditionalContent with
 * \ref ConditionalContentWriter::writeConditionalContent.
```

File: Cesium3DTilesWriter/include/Cesium3DTilesWriter/ConditionalContentWriter.h [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  std::vector<std::string> errors;
+
+  /**
+   * @brief Warnings, if any, that occurred during the write process.
+   */
+  std::vector<std::string> warnings;
+};
+
+/**
+ * @brief Options for how to write a ConditionalContent.
```
Comment (j9liu) [inline Cesium3DTilesWriter/include/Cesium3DTilesWriter/ConditionalContentWriter.h]: Might be nice to `\ref` the `ConditionalContent` mentions if it's not too much trouble.

File: Cesium3DTilesWriter/src/ConditionalContentWriter.cpp [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+ConditionalContentWriter::writeConditionalContent(
+    const Cesium3DTiles::ConditionalContent& conditionalContent,
+    const ConditionalContentWriterOptions& options) const {
+  CESIUM_TRACE("ConditionalContentWriter::writeConditionalContent");
+
+  const CesiumJsonWriter::ExtensionWriterContext& context =
+      this->getExtensions();
+
+  ConditionalContentWriterResult result;
+  std::unique_ptr<CesiumJsonWriter::JsonWriter> writer;
```
Comment (j9liu) [inline Cesium3DTilesWriter/src/ConditionalContentWriter.cpp]: Per style guide:

```suggestion
  std::unique_ptr<CesiumJsonWriter::JsonWriter> pWriter;
```

**Review by j9liu — CHANGES_REQUESTED** [review]
Thanks @lilleyse! Just a few minor comments to address before merge.

File: tools/generate-classes/package.json (hunk 1 above)
Comment (lilleyse) [reply tools/generate-classes/package.json:8]: The extra input is needed so that it generates code for `conditionContent.schema.json` which is a standalone file like a subtree or schema.

`3DTILES_content_conditional` is already merged into `cesium-native` but this extra input is still necessary.

File: Cesium3DTiles/generated/include/Cesium3DTiles/Extension3dTilesContentConditional.h (hunk 2 above)
Comment (lilleyse) [reply Cesium3DTiles/generated/include/Cesium3DTiles/Extension3dTilesContentConditional.h]: Posted in https://github.com/CesiumGS/3d-tiles/pull/834/changes#r3191077719

File: Cesium3DTilesWriter/include/Cesium3DTilesWriter/ConditionalContentWriter.h (hunk 4 above)
Comment (lilleyse) [reply Cesium3DTilesWriter/include/Cesium3DTilesWriter/ConditionalContentWriter.h]: Looks like the style guide suggests `@ref` so I'll update to that

(`\ref`, `@link` and `@ref` all seem to be used throughout the code, we might need to do a cleanup pass)

File: Cesium3DTilesWriter/include/Cesium3DTilesWriter/ConditionalContentWriter.h (hunk 5 above)
Comment (lilleyse) [reply Cesium3DTilesWriter/include/Cesium3DTilesWriter/ConditionalContentWriter.h]: I used `@ref` in the class descriptions and "conditional content" in the member descriptions. That felt like a good balance otherwise there would have been a lot of `@ref Cesium3DTiles::CondtionalContent`.

File: Cesium3DTiles/generated/include/Cesium3DTiles/Extension3dTilesContentConditional.h (hunk 2 above)
Comment (lilleyse) [reply Cesium3DTiles/generated/include/Cesium3DTiles/Extension3dTilesContentConditional.h]: Fixed in https://github.com/CesiumGS/cesium-native/pull/1351/commits/af089b71cb32483e016846e82306e1418bd7d36a

File: Cesium3DTiles/generated/include/Cesium3DTiles/Extension3dTilesContentConditionalDimensionsValue.h (hunk 3 above)
Comment (lilleyse) [reply Cesium3DTiles/generated/include/Cesium3DTiles/Extension3dTilesContentConditionalDimensionsValue.h:33]: Fixed in https://github.com/CesiumGS/cesium-native/pull/1351/commits/af089b71cb32483e016846e82306e1418bd7d36a

**Comment by lilleyse** [comment]
@j9liu ready for another review!

**Review by j9liu — APPROVED** [review]
Thanks @lilleyse!
